BE/src/main/java/louie/dong/airbnb/accommodation/AccommodationService.java

```diff
@@ -0,0 +1,116 @@
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+public class AccommodationService {
+
+	private final AccommodationRepository accommodationRepository;
+	private final WishRepository wishRepository;
+
+	public AccommodationPriceResponse findPrices(String country) {
+		List<Integer> prices = accommodationRepository.findByAccommodationPrices(country);
+		return new AccommodationPriceResponse(calculateAverage(prices), prices);
+	}
+
+	public AccommodationDetailResponse findById(Long id) {
+		Accommodation accommodation = getAccommodation(id);
+		boolean wish = wishRepository.existsByAccommodationId(id);
+		return new AccommodationDetailResponse(accommodation, wish);
+	}
+
+	public AccommodationDetailPriceResponse findDetailPrice(Long id,
+		AccommodationDetailPriceRequest accommodationDetailPriceRequest) {
+		Accommodation accommodation = getAccommodation(id);
+		LocalDate checkIn = accommodationDetailPriceRequest.getCheckIn();
+		LocalDate checkOut = accommodationDetailPriceRequest.getCheckOut();
+
+		int date = (int) checkIn.until(checkOut, ChronoUnit.DAYS);
+		int totalPrice = accommodation.getPrice() * date;
+		double discountRate = getDiscountRate(date) * 0.01;
+		int discountPrice = (int) (totalPrice * discountRate);
+		int finalPrice = totalPrice - discountPrice + accommodation.getCleaningFee()
+			+ accommodation.getServiceFee() + accommodation.getAccommodationFee();
+
+		return new AccommodationDetailPriceResponse(accommodation.getPrice(), date, totalPrice,
+			WEEKLY.getDiscountRate(), discountPrice, accommodation.getCleaningFee(),
+			accommodation.getServiceFee(), accommodation.getAccommodationFee(), finalPrice);
+	}
+
+	public AccommodationSearchResponse findAccommodations(
+		AccommodationSearchRequest accommodationSearchRequest) {
+		LocalDate checkIn = accommodationSearchRequest.getCheckIn();
+		LocalDate checkOut = accommodationSearchRequest.getCheckOut();
+
+		List<Accommodation> accommodations = accommodationRepository.searchAccommodations(
+			accommodationSearchRequest.getCountry(),
+			LocalDateTime.of(checkIn, LocalTime.of(0, 0)),
+			LocalDateTime.of(checkOut, LocalTime.of(0, 0)),
+			accommodationSearchRequest.getMinPrice(), accommodationSearchRequest.getMaxPrice(),
+			accommodationSearchRequest.getGuestCount());
+
+		List<AccommodationResponse> accommodationResponses = createAccommodationResponses(
+			accommodations, checkIn, checkOut);
+		return new AccommodationSearchResponse(accommodations.size(), accommodationResponses);
+	}
+
+	private List<AccommodationResponse> createAccommodationResponses(
+		List<Accommodation> accommodations, LocalDate checkIn, LocalDate checkOut) {
+		int stayNight = (int) ChronoUnit.DAYS.between(checkIn, checkOut);
+		return accommodations.stream()
+			.map(accommodation -> new AccommodationResponse(accommodation,
+				stayNight * accommodation.getPrice(), accommodation.existsWish()))
```

Comment:
AccommodationResponse의 생성자 또는 팩토리 메소드를 이용해서 개선해봅시다.
```suggestion
			.map(accommodation -> new AccommodationResponse(accommodation, nights))
```

Reply:
오 이렇게 하니 코드가 훨씬 간결해지네요 ㅎㅎ 
왕민 말씀대로 적용해보겠습니다 😀